backend/hidrive/hidrive.go

```diff
@@ -0,0 +1,962 @@
+			Default:  defaultUploadCutoff,
+			Advanced: true,
+		}, {
+			Name: "upload_concurrency",
+			Help: `Concurrency for chunked uploads.
+
+This is the upper limit for how many transfers for the same file are running concurrently.
+Setting this above to a value smaller than 1 will cause uploads to deadlock.
+
+If you are uploading small numbers of large files over high-speed links
+and these uploads do not fully utilize your bandwidth, then increasing
+this may help to speed up the transfers.`,
+			Default:  defaultUploadConcurrency,
+			Advanced: true,
+		}, {
+			Name:     config.ConfigEncoding,
+			Help:     config.ConfigEncodingHelp,
+			Advanced: true,
+			// HiDrive only supports file or folder names of 255 characters or less.
+			// Names containing "/" are not supported.
+			// The special names "." and ".." are not supported.
+			Default: (encoder.EncodeZero |
+				encoder.EncodeSlash |
+				encoder.EncodeDot),
+		}}...),
+	})
+}
+
+// Options defines the configuration for this backend.
+type Options struct {
+	EndpointAPI                  string               `config:"endpoint"`
+	OptionalMemberCountDisabled  bool                 `config:"disable_fetching_member_count"`
+	UnicodeNormalizationDisabled bool                 `config:"disable_unicode_normalization"`
+	UploadChunkSize              fs.SizeSuffix        `config:"chunk_size"`
+	UploadCutoff                 fs.SizeSuffix        `config:"upload_cutoff"`
+	UploadConcurrency            int64                `config:"upload_concurrency"`
+	Enc                          encoder.MultiEncoder `config:"encoding"`
+	RootPrefix                   string               `config:"root_prefix"`
+	ScopeAccess                  string               `config:"scope_access"`
+	ScopeRole                    string               `config:"scope_role"`
+}
+
+// Fs represents a remote hidrive.
+type Fs struct {
+	name         string           // name of this remote
+	root         string           // the path we are working on
+	opt          Options          // parsed options
+	features     *fs.Features     // optional features
+	srv          *rest.Client     // the connection to the server
+	pacer        *fs.Pacer        // pacer for API calls
+	retrier      *pacer.Pacer     // pacer with no delays to retry certain operations once
+	tokenRenewer *oauthutil.Renew // renew the token on expiry
+}
+
+// Object describes a hidrive object.
+//
+// Will definitely have the remote-path but may lack meta-information.
+type Object struct {
+	fs          *Fs       // what this object is part of
+	remote      string    // The remote path
+	hasMetadata bool      // whether info below has been set
+	size        int64     // size of the object
+	modTime     time.Time // modification time of the object
+	id          string    // ID of the object
+	hash        string    // content-hash of the object
+}
+
+// ZeroDelayCalculator is a pacer.Calculator that never delays.
```

Comment:
The `hidrive`-backend uses the `fs.pacer`-mechanism to retry some operations once.
(That is the `retrier`-pacer of the `Fs`-implementation.)

For retrying those operations no delays are needed; the same could be accomplished using a for-Loop or some duplicate code.
As no delay is needed, the `ZeroDelayCalculator` is introduced for this pacer.

If considered useful in general, the `ZeroDelayCalculator` could also be moved to `lib/pacer/parcers.go`.

Reply:
@ncw Do you want `ZeroDelayCalculator` in `lib/pacer` ?